app/bin/service/analyzer.dart

```diff
@@ -28,16 +28,17 @@ Future main() async {
 
   withAppEngineServices(() async {
     _initFlutterSdk().then((_) async {
-      final ReceivePort mainReceivePort = new ReceivePort();
       final ReceivePort errorReceivePort = new ReceivePort();
 
       Future startIsolate() async {
         logger.info('About to start analyzer isolate...');
+        final ReceivePort mainReceivePort = new ReceivePort();
         await Isolate.spawn(
           _runScheduler,
           [mainReceivePort.sendPort],
           onError: errorReceivePort.sendPort,
           onExit: errorReceivePort.sendPort,
+          errorsAreFatal: true,
         );
         final List<SendPort> sendPorts = await mainReceivePort.take(1).toList();
         registerTaskSendPort(sendPorts[0]);
```

Comment:
Since we know ahead-of-time the number of isolates, it might make sense to shard the stream of `Task`s.

i.e. letting isolate `isolateId` only analyze packages `"$name/$version".hashCode % #num-isolates == isolateId`. Otherwise the isolates might be traversing the very same stream of `Task`s and analyze them, just to figure out that another isolate was faster.